src/test/java/de/tum/in/www1/artemis/util/UserTestService.java

```diff
@@ -148,7 +149,7 @@ public void updateUser_asAdmin_isSuccessful() throws Exception {
         final var newLastName = "Wayne";
         final var newImageUrl = "foobar.png";
         final var newLangKey = "DE";
-        final var newAuthorities = Set.of(Role.TEACHING_ASSISTANT.getAuthority()).stream().map(authorityRepository::findById).filter(Optional::isPresent).map(Optional::get)
+        final var newAuthorities = Stream.of(Role.TEACHING_ASSISTANT.getAuthority()).map(authorityRepository::findById).filter(Optional::isPresent).map(Optional::get)
                 .collect(Collectors.toSet());
```

Comment:
```suggestion
        final var newAuthorities = Stream.of(Role.TEACHING_ASSISTANT.getAuthority()).map(authorityRepository::findById).flatMap(Optional::stream).collect(Collectors.toSet());
```
The `.filter(Optional::isPresent).map(Optional::get)` can be combined into a single `flatMap(Optional::stream)`.

(Maybe a bit of personal preference, feel free to ignore the suggestion.)

Reply:
I'll keep that since I think changing this would hinder readability of the code. Filter followed by map shows directly that only present items will be used, you need to know that this is done implicitly when using `flatMap(Optional::stream)` (which I didn't up to this point).